src/app-layer-htp.c

```diff
@@ -4384,15 +4086,12 @@ libhtp:\n\
 
     void *user_data = NULL;
     (void)SCRadixFindKeyIPV4BestMatch((uint8_t *)f->dst.addr_data32, cfgtree, &user_data);
-    if (user_data != NULL) {
-        HTPCfgRec *htp_cfg_rec = user_data;
-        htp = htp_cfg_rec->cfg;
-        SCLogDebug("LIBHTP using config: %p", htp);
-    }
-    if (htp == NULL) {
-        printf("Could not get config for: %s\n", addr);
-        goto end;
-    }
+    FAIL_IF_NULL(user_data);
```

Comment:
As the tests passed, I suppose there's no harm here. But wanted to point out that this changes the original format a bit, since we don't jump to `end` in this case, even if `user_data` was `NULL`